drivers/ieee802154/ieee802154_dw1000.c

```diff
@@ -769,12 +769,66 @@ static int dwt_filter(const struct device *dev,
 	return -ENOTSUP;
 }
 
+// From 7.2.31.1 from User Manual
+#define DWT_TX_POWER_CTRL_OFF 7
+#define DWT_TX_POWER_CTRL_COARSE_MASK 0xE0U
+#define DWT_TX_POWER_CTRL_COARSE_STEPS 2.5
+#define DWT_TX_POWER_CTRL_FINE_MASK 0x1FU
+#define DWT_TX_POWER_CTRL_FINE_STEPS 0.5
+
+#define DWT_TX_POWER_TXPOWPHR_SHIFT 8
+#define DWT_TX_POWER_TXPOWSD_SHIFT 16
+
+// These should go into DT probably
+#define DWT_TX_POWER_BOOSTP500_MOD 3
+#define DWT_TX_POWER_BOOSTP250_MOD 6
+#define DWT_TX_POWER_BOOSTP125_MOD 9
+#define DWT_TX_POWER_PHR_MOD 0
+
+uint8_t dwt_encode_power_tx(float power) {
+	uint8_t coarse = DWT_TX_POWER_CTRL_OFF;
+	uint8_t fineTuneVal = 0;
+	if (power >= 0) {
```

Comment:
In this case I believe an ASSERT would not work because it is considered that a negative power will use the DWT_TX_POWER_CTRL_OFF value since no negative values can be used according to the documentation. 
Should maybe I explicitly write that as a comment?

Reply:
Fixed